internal/plugins/openshift/v1/plugin.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2021 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"sigs.k8s.io/kubebuilder/v2/pkg/model/config"
+	"sigs.k8s.io/kubebuilder/v2/pkg/plugin"
+
+	"github.com/operator-framework/operator-sdk/internal/plugins"
+)
+
+const (
+	pluginName = "openshift" + plugins.DefaultNameQualifier
+)
+
+var (
+	pluginVersion   = plugin.Version{Number: 1}
+	pluginConfigKey = plugin.Key(pluginName, pluginVersion.String())
+)
+
+// Config configures this plugin, and is saved in the project config file.
+type Config struct{}
+
+// RunInit updates a newly initialized project with OpenShift-specific configuration.
+func RunInit(cfg *config.Config) error {
+
+	if err := runInit(cfg); err != nil {
+		return err
+	}
+
+	// Update the plugin config section with this plugin's configuration.
+	pCfg := Config{}
+	if err := cfg.EncodePluginConfig(pluginConfigKey, pCfg); err != nil {
+		return fmt.Errorf("error writing plugin config for %s: %v", pluginConfigKey, err)
+	}
+
+	return nil
+}
+
+// imageSubstitutions is a map of paths to image substitutions.
+var imageSubstitutions = map[string]struct {
+	fromTagRE *regexp.Regexp
+	toTag     string
+}{
+	filepath.Join("config", "default", "manager_auth_proxy_patch.yaml"): {
+		regexp.MustCompile("gcr.io/kubebuilder/kube-rbac-proxy:v[0-9]+\\.[0-9]+\\.[0-9]+"),
+		"quay.io/openshift/origin-kube-rbac-proxy:4.7.0",
+	},
+}
+
+func runInit(cfg *config.Config) error {
+
+	for filePath, subst := range imageSubstitutions {
+		b, err := ioutil.ReadFile(filePath)
+		if err != nil {
+			return fmt.Errorf("error reading file to substitute %q: %v", subst.toTag, err)
+		}
+		info, err := os.Stat(filePath)
+		if err != nil {
+			return fmt.Errorf("error stating file to substitute %q: %v", subst.toTag, err)
+		}
+		// Ensure replacements occur, and error out if they haven't. This should be caught
+		// by e2e tests.
+		numReplacements := 0
```

Comment:
This may be redundant, since testdata will reflect whether the replacement occurred.